File: tests/fixtures/codegen/dimensions/pointers-expected/queries/arrays/updateflexible.go
package arrays

import (
	"context"
	pgsidpgx "example.com/pgsid-validation/generated/queries/pgsid/pgx"
	"github.com/jackc/pgx/v5/pgtype"
)

const UpdateFlexibleSQL = "UPDATE arrays SET flexible = $1, optional_flexible = $2 WHERE id = $3;"

type UpdateFlexibleParams struct {
	Flexible pgtype.Array[int32]  `db:"flexible"`
	Optional *pgtype.Array[int32] `db:"optional"`
	Id       *int32               `db:"id"`
}

func (q *Queries) UpdateFlexible(ctx context.Context, params UpdateFlexibleParams) (int64, error) {
	arrayInput1, err := pgsidpgx.ValidateArrayInput[int32](params.Flexible, []int{1, 2}, "UpdateFlexible", "flexible", false)
	if err != nil {
		return 0, err
	}
	arrayInput2, err := pgsidpgx.ValidateArrayInput[int32](params.Optional, []int{1, 2}, "UpdateFlexible", "optional", true)
	if err != nil {
		return 0, err
	}
	tag, err := q.db.Exec(ctx, UpdateFlexibleSQL, arrayInput1, arrayInput2, params.Id)
	if err != nil {
		return 0, err
	}
	return tag.RowsAffected(), nil
}

File: tests/fixtures/codegen/dimensions/structs-expected/typescript/runtime/queries/arrays/UpdateFlexible.ts
import type { UpdateFlexibleParams } from "../../../types/queries/arrays/UpdateFlexible.js";
export type { UpdateFlexibleParams } from "../../../types/queries/arrays/UpdateFlexible.js";
import type * as pgsid from "./pgsid/queryable.js";
export const updateFlexibleSql = "UPDATE arrays SET flexible = $1, optional_flexible = $2 WHERE id = $3;";
export async function updateFlexible(db: pgsid.Queryable, params: UpdateFlexibleParams): Promise<number> {
    const result = await db.query(updateFlexibleSql, [params["flexible"], params["optional"], params["id"]]);
    return result.rowCount ?? 0;
}
